src/node/controller/messages.rs

```diff
@@ -136,9 +137,53 @@ impl Controller {
     pub async fn handle_publish(
         &mut self,
         _header: &Header,
-        _publish: Publish,
+        publish: Publish,
     ) -> anyhow::Result<()> {
         // dbg!(publish);
+        let _block = match publish.block_holder {
+            BlockHolder::Send(_) => {
+                todo!("Received a send block")
+            }
+            BlockHolder::Receive(_) => {
+                todo!("Received a receive block")
+            }
+            BlockHolder::Open(_) => {
+                todo!("Received an open block")
+            }
+            BlockHolder::Change(_) => {
+                todo!("Received a change block")
+            }
+            BlockHolder::State(state_block) => {
+                let mut block = Block::from_state_block(&state_block);
+                let block_hash: &BlockHash = {
+                    block.calc_hash().unwrap();
```

Comment:
Please use `context()?` or `with_context()?` instead of unwrap so we can handle problems gracefully instead of panicking.

Reply:
Ah right. Btw I'm starting to be more convinced about requiring a hash to create a valid block. The only reason we put it separately is in case it fails, but I quite sure that a hash can't fail per se, so if it fails it's a programming defect (unrecoverable error in Rust lingo). In other words, I don't think it makes sense to keep a block around incapable of having a hash.

Reply:
OK, if you're convinced go ahead!